### ISSUE TITLE
Export as PNG with X frames per row.

### ISSUE THREAD

**Issue by RainingChain** [juliandescottes/piskel#293]
Right now, you can only export as PNG with all frames side by side.

It would be great if one could specify the amount of frames per row. This could be done by adding a numeric input next to the 'Download PNG' button.

Here is a function that converts an image with all frames side by side to an image with the amount of rows specified.

```javascript
/*
  img:Image, image with all frames side by side
  totalFrame:Number, total amount of frames in the image
  framesPerRow:Number,
  return Image, image with the right amount of frames per row
*/
var convertImage = function(img,totalFrame,framesPerRow){
	if(framesPerRow === totalFrame)
		return img;
	
	var frameWidth = img.width/totalFrame;
	var frameHeight = img.height;
	var rowCount = Math.ceil(totalFrame/framesPerRow);
	
	var finalImgWidth = frameWidth * framesPerRow;
	var finalImgHeight = frameHeight * rowCount;
	
	var canvas = document.createElement('canvas');
	canvas.width = finalImgWidth;
	canvas.height = finalImgHeight;
	
	var ctx = canvas.getContext('2d');
	
	for(var i = 0; i < rowCount; i++){
		var cropWidth = finalImgWidth;
		if(i * finalImgWidth + finalImgWidth > img.width)	//incase row is not full (framesPerRow is not a factor of totalFrame)
			cropWidth = img.width - i * finalImgWidth;
		
		ctx.drawImage(img,
			i * finalImgWidth,0,
			cropWidth,frameHeight,
			0,i * frameHeight,
			cropWidth,frameHeight
		);
	}
	var newImg = new Image();
	newImg.src = canvas.toDataURL('image/png');
	return newImg;
}
```

**Comment by bsmith555** [comment]
Please also consider an option for specifying the maximum pixel width of export file, or perhaps even both maximum dimensions.

**Comment by LuciusSixPercent** [comment]
Maybe there could be an "advanced options" in the export dialog where you could specify rows, columns, dimensions, margins, padding, etc...

And I may be going too far here, but perhaps and option to export an XML with some metadata like frame duration and other custom tags. That could be a life changer for some! Although that may require changing how frames work internally and all... one can dream right?

**Comment by juliandescottes** [comment]
Looks like there is a lot of interest for improvements to the import/export, so I will get started on that next.

@LuciusSixPercent : I like what you described with the XML, but is there some kind of standard widely recognized by game engines or should it be heavily customisable (XML, json etc ...)

**Comment by bsmith555** [comment]
@juliandescottes It's a bit of a non-answer from me, but Unity and GameMaker don't need a metadata file. They both have tools for slicing sprite sheets built in.

**Comment by Tajnymag** [comment]
Could you implement a simple X*Y frames export and add the Metadata support later? As bsmith has said, many engines don't need them anymore.

**Comment by juliandescottes** [comment]
@bsmith555 @Tajnymag 
> Could you implement a simple X*Y frames export and add the Metadata support later?

Sounds good to me. 
What is the most meaningful input for the export : width & height in pixels OR rows & cols in "number of frames" ?

**Comment by Tajnymag** [comment]
@juliandescottes I would prefer the rows and cols as it seems more useful

**Comment by Tajnymag** [comment]
@juliandescottes Are we there yet? :)

**Comment by juliandescottes** [comment]
@Tajnymag not yet sorry :(

**Comment by juliandescottes** [comment]
Fixed with #473 

![piskel-export-spritesheet](https://cloud.githubusercontent.com/assets/1141550/15632498/20f743ee-2594-11e6-8bdd-bcb336745f14.gif)
